model/src/gateway/payload/update_status.rs

```diff
@@ -11,48 +62,71 @@ pub struct UpdateStatus {
 }
 
 impl UpdateStatus {
+    /// Create a new, valid [`UpdateStatus`] payload.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error of type [`UpdateStatusErrorType::MissingActivity`] if
+    /// an empty set of activites is provided.
     pub fn new(
-        activities: impl Into<Option<Vec<Activity>>>,
+        activities: impl Into<Vec<Activity>>,
         afk: bool,
         since: impl Into<Option<u64>>,
         status: impl Into<Status>,
-    ) -> Self {
-        Self {
-            d: UpdateStatusInfo::new(activities, afk, since, status),
+    ) -> Result<Self, UpdateStatusError> {
+        let d = UpdateStatusInfo::new(activities, afk, since, status)?;
+
+        Ok(Self {
+            d,
             op: OpCode::StatusUpdate,
-        }
+        })
     }
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
 pub struct UpdateStatusInfo {
-    pub activities: Option<Vec<Activity>>,
+    /// User's activities.
+    ///
+    /// At least one is required.
+    pub activities: Vec<Activity>,
     pub afk: bool,
     pub since: Option<u64>,
     pub status: Status,
 }
 
 impl UpdateStatusInfo {
+    /// Create a new, valid [`UpdateStatusInfo`] struct.
```

Comment:
I'm not sure what the best way to fix this is but it reads a bit weird. Maybe "Create a validated status update payload"? `Info` isn't the best name suffix here, admittedly.

Reply:
Info can be changed when we rename to UpdatePresence